<b><center>Taking Input from the user</b></center>

In this repository we'll learn how to take imput from the user. This is very easy. There are two options to take the input, first is declaring the variables and second is to take the input from the user.

- Declaring the variables

In [1]:
a = 20
b = 7
print(a+b)

27


In [2]:
a = 20
b = 7
c = a + b
print(c)

27


- Taking the input from the user

In [3]:
a = input("Enter the first number: ")
b = input("Enter the second number: ")
c = a + b
print("Result is: ",c)

Enter the first number:  20
Enter the second number:  7


Result is:  207


<b>Why is the answer 207 instead of 27?</b>

The above code just contatenated the values. It didn't quite add the values. So what we should do? <br>
Here we use a concept known as <b>Typecasting</b>

Type Casting is the method to convert the variable data type into a certain data type in order to the operation required to be performed by users. <br>
We usually type cast the input to integer, float or string by specifying the input() function inside the type. <br>

It is used in computer programming to ensure variables are correctly processed by a function.

It is used in computer programming to ensure variables are correctly processed by a function. <br>
An example of typecasting is converting an integer to a string. 

<b> When do you need to use type-conversions? <b>

- The value to be stored is in a form not supported by that data type
- To reduce the memory in use, relevant to the value
- The value to be stored is beyond the max limit

Python has two types of type conversion:
- Implicit Type casting
- Explicit Type casting

<b> Implicit Type Casting <b>

Implicit type conversion is performed automatically by the interpreter, without user intervention. Python automatically converts one data type to another data type. This process doesn’t need any user involvement Python promotes the conversion of lower data type.

In [4]:
a = 5
b = 6.0
c = a+b
print(c)

11.0


You see here int is converted to the higher data type float to avoid the loss of data. This type of conversion or type casting is called Up Casting. <br>
Whereas, when higher data type is converted to lower data type, it is called Down Casting.

<b>Explicit Type Casting</b>

In this method, Python needs the user involvement to convert the variable data type into certain data type in order to the operation required. <br>

Mainly, type casting can be done with these data type function: <br>

- int() : int() function take float or string as an argument and return int type object.
- float() : float() function take int or string as an argument and return float type object.
- str() : str() function take float or int as an argument and return string type object.

In [5]:
a = '5' #String Data Type
print(int(a))

5


In [6]:
a = 5.7 #Float Data Type
print(int(a))

5


In [7]:
a = 25 #Int Data type
print(float(a))

25.0


In [8]:
a = 16
b = str(a)
print(b)
print(type(b))

16
<class 'str'>


In [9]:
c = "5.9" #String Data Type
d= float(c)
 
print(d)
print(type(d))

5.9
<class 'float'>


So as to solve the above problem of 207 we use typecasting like this: <br>

In [10]:
a = int(input("Enter the first number: "))
b = int(input("Enter the second number: "))
c = a + b
print("Result is: ",c)

Enter the first number:  20
Enter the second number:  7


Result is:  27


In [11]:
a = float(input("Enter the first number: "))
b = float(input("Enter the second number: "))
c = a + b
print("Result is: ",c)

Enter the first number:  5.8
Enter the second number:  5.2


Result is:  11.0
